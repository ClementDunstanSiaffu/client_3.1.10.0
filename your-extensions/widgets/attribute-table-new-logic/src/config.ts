import { ImmutableObject } from 'seamless-immutable'

export interface Config {
  tabEmptyNotShow: boolean,
  buttonFilter: boolean,
  downloadCGI: boolean,
  downloadECGI: boolean,
  downloadKMZ: boolean
}

export type IMConfig = ImmutableObject<Config>
