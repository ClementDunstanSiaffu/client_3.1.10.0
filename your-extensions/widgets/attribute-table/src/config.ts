import { ImmutableObject } from 'seamless-immutable'

export interface Config {
  "buttonFilter": boolean,
  "downloadCGI": boolean,
  "downloadECGI": boolean,
  "downloadKMZ": boolean
}

export type IMConfig = ImmutableObject<Config>
