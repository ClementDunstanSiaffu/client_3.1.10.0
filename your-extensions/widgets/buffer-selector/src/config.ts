import { ImmutableObject } from 'seamless-immutable'

export interface Config {
  checkUrlExternal:boolean,
  urlConfig:string,

  idWidgetTable: string,
  servizi_da_mappa: boolean,
  maxWidthBuffer : number,
  slideBuffer : boolean,
  queryConGeometry : boolean,
  listvalues: ListValue[]
}
export interface ListValue {
  name: string,
  listIds: [],
  link: string
}


export type IMConfig = ImmutableObject<Config>
