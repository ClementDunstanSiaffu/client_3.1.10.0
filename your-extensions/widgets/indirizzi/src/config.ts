import { ImmutableObject } from 'seamless-immutable'

export interface Config {
  settings:{
    idWidgetTable:string,
    exampleConfigProperty: string,
    servizi_da_mappa:boolean,
    maxWidthBuffer : number,
    slideBuffer :boolean,
    queryConGeometry : boolean,
  },
  services:{
    service_1: {
      "url":string,
      "layerListIds":number[],
      "title":string
    },
    service_2: {
      "url":string,
      "layerListIds":number[],
      "title":string
    },
    service_3: {
      "url":string,
      "layerListIds":number[],
      "title":string
    },
    service_4: {
      "url":string,
      "layerListIds":number[],
      "title":string
    }
  }

}

export type IMConfig = ImmutableObject<Config>
