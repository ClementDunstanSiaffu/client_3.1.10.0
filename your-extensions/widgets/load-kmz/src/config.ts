import { ImmutableObject } from 'seamless-immutable'

export interface Config {
  idWidgetTable:string,
  exampleConfigProperty: string,
  listvalues: ListValue[]
}
export interface ListValue {
  name: string,
  listIds: [],
  link: string
}

export type IMConfig = ImmutableObject<Config>
