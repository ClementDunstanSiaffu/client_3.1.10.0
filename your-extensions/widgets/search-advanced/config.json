{
  "searchItems": {
    "url": "https://tsarcgis.aqp.it/server/rest/services/Operational/QUERY_SIT/FeatureServer/0",
    "queryWhere": "NOMECOMUNE is not null",
    "outFields": ["OBJECTID","NOMECOMUNE","ISTAT"],
    "outFieldsDisplay": ["*"]
  },
  "searchAmbiti": {
    "url": "https://tsarcgis.aqp.it/server/rest/services/Operational/QUERY_SIT/FeatureServer/1",
    "queryWhere": "NOMEAMBITO is not null",
    "outFields": ["IDAMBITO","NOMEAMBITO"]
  },
  "searchSTO": {
    "url": "https://tsarcgis.aqp.it/server/rest/services/Operational/QUERY_SIT/FeatureServer/2",
    "queryWhere": "NOMECOMPARTIMENTO is not null",
    "outFields": ["IDCOMPARTIMENTO ","NOMECOMPARTIMENTO"]
  }
}
