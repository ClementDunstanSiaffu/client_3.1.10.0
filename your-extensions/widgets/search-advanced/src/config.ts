import { ImmutableObject } from 'seamless-immutable'

export interface Config {
  "idWidgetTable":string,
  "searchItems": {
    "url": string,
    "queryWhere": string,
    "outFields": string[],
    "outFieldsDisplay": string[],
  },
  "searchAmbiti": {
    "url": string,
    "queryWhere": string,
    "outFields": string[]
  },
  "searchSTO": {
    "url": string,
    "queryWhere": string,
    "outFields": string[]
  }
}

export type IMConfig = ImmutableObject<Config>
