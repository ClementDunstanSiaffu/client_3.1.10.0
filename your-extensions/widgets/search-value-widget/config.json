
{
    "service": {
        "url": "https://services.arcgis.com/P3ePLMYs2RVChkJx/ArcGIS/rest/services/Bureau_of_Economic_Analysis_Gross_Domestic_Product_GDP_2019/FeatureServer"
    }
}