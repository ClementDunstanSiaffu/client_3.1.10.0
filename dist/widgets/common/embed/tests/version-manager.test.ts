import { Immutable } from 'jimu-core'
import { versionManager } from '../src/version-manager'
import { EmbedType } from '../src/config'
import { withEmptyExp, withoutEmptyExp } from './expression-const'

let upgrader = null

jest.mock('jimu-core', () => {
  return {
    ...jest.requireActual<any>('jimu-core'),
    utils: {
      getUUID: jest.fn(() => 'e2f5')
    }
  }
})

describe('Test embed version-manager for version 1.4.0', () => {
  beforeAll(() => {
    upgrader = versionManager.versions?.filter(function (version) {
      return version.version === '1.4.0'
    })[0]?.upgrader
  })

  it('should return new config', () => {
    const oldConfig = Immutable({
      embedType: EmbedType.Url,
      embedCode: '',
      staticUrl: '',
      expression: {
        name: 'Expression 1',
        parts: [{
          dataSourceId: 'ds_9212',
          exp: '{OBJECTID}',
          isFromRepeatedDataSourceContext: false,
          jimuFieldName: 'OBJECTID',
          type: 'FIELD'
        }, {
          exp: '+',
          type: 'OPERATOR'
        }, {
          exp: '666',
          type: 'NUMBER'
        }, {
          exp: '+',
          type: 'OPERATOR'
        }, {
          exp: '"search"',
          type: 'STRING'
        }]
      },
      autoRefresh: false,
      autoInterval: 10
    })
    const newConfig = upgrader(oldConfig)
    expect(newConfig).toMatchSnapshot()
  })
})

describe('Test embed version-manager for version 1.7.0', () => {
  beforeAll(() => {
    upgrader = versionManager.versions?.filter(function (version) {
      return version.version === '1.7.0'
    })[0]?.upgrader
  })

  it('remove empty tag of expression', () => {
    const oldConfig = Immutable({
      embedType: EmbedType.Url,
      expression: withEmptyExp,
      autoRefresh: false,
      autoInterval: 10
    })
    const newConfig = upgrader(oldConfig)
    expect(newConfig.expression).toBe(withoutEmptyExp)
  })
})
