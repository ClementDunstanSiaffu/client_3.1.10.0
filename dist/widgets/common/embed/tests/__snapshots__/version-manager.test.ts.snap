// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test embed version-manager for version 1.4.0 should return new config 1`] = `
Object {
  "autoInterval": 10,
  "autoRefresh": false,
  "embedCode": "",
  "embedType": "url",
  "enableLabel": true,
  "expression": "<p><exp data-uniqueid=\\"e2f5\\" data-dsid=\\"ds_9212\\" data-expression=\\"%7B%22name%22%3A%22%7BOBJECTID%7D%22%2C%22parts%22%3A%5B%7B%22dataSourceId%22%3A%22ds_9212%22%2C%22exp%22%3A%22%7BOBJECTID%7D%22%2C%22isFromRepeatedDataSourceContext%22%3Afalse%2C%22jimuFieldName%22%3A%22OBJECTID%22%2C%22type%22%3A%22FIELD%22%7D%5D%7D\\">{OBJECTID}</exp>666search</p>",
  "label": "Expression 1",
  "staticUrl": "",
}
`;
